src/libPMacc/include/nvidia/atomic.hpp

```diff
@@ -34,40 +36,113 @@ namespace PMacc
 namespace nvidia
 {
 
+    namespace detail {
+
+        template<typename T_Type, bool T_isKepler>
+        struct AtomicAllInc
+        {
+            DINLINE T_Type
+            operator()(T_Type* ptr)
+            {
+                return atomicAdd(ptr, 1);
+            }
+        };
+
+#if PMACC_CUDA_ARCH >= 300
```

Comment:
They need to be. They use 3.00 features and having only the above struct outside of it hides everything related to it. It makes it  pretty clear that only for 3.00 ISAs optimized stuff exists. Nothing wrong with that IMO